### ISSUE TITLE
ttnn.tanh low PCC for a specific input used in albert_v2_base for masked lm

### ISSUE THREAD

**Issue by amalbasaTT** [tenstorrent/tt-metal#17721]
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/pytorch2.0_ttnn/issues/739)

**Describe the bug**
ttnn.tanh has low PCC for a specific input used in albert_v2_base for masked lm. Happens at `ttnn_tanh = ttnn.tanh(ttnn_multiply_4, )`. PCC is 0.9945167693107032.
Input to ttnn.tanh has spec:
- input_values: [tanh_input.csv](https://github.com/user-attachments/files/18707946/tanh_input.csv)
- shape: (1, 9, 3072)
- dtype: ttnn.bfloat16
- dlayout: ttnn.TILE_LAYOUT

**To Reproduce**
1. Install torch-ttnn as explained in https://github.com/tenstorrent/pytorch2.0_ttnn?tab=readme-ov-file 
2. Possition to pytorch2.0_ttnn directory
3. Generate `albert-base-v2_code.py` and `albert-base-v2_inputs.pickle` by running command:
```
pytest tests/models/albert/test_albert_masked_lm.py --gen_op_accuracy_tests
```
4. In `albert-base-v2_code.py`, after `ttnn_tanh = ttnn.tanh(ttnn_multiply_4, )`, add `test_accuracy(tanh, ttnn_tanh)`. In function definition `test_accuracy`, change pcc to be 0.999.
5. Run `albert-base-v2_code.py`:
```
python3 tests/autogen_accuracy_tests/albert-base-v2_code.py
```
Low pcc error will be at `test_accuracy(tanh, ttnn_tanh)`

**Additional Content**
Functions to extract and load inputs to `ttnn_tanh`:
```
def save_tensor_to_csv(tensor, filename):
    import pandas as pd
    tensor = tensor.contiguous()
    if tensor.dtype == torch.bfloat16:
        tensor = tensor.to(torch.float32)
    if tensor.ndim == 0: 
        df = pd.DataFrame([tensor.item()])
    elif tensor.ndim == 1:
        df = pd.DataFrame(tensor.numpy())
    elif tensor.ndim == 2:
        df = pd.DataFrame(tensor.numpy())
    elif tensor.ndim == 3:
        df = pd.DataFrame(tensor.view(-1, tensor.size(-1)).numpy())
    elif tensor.ndim == 4: 
        df = pd.DataFrame(tensor.view(-1, tensor.size(-1)).numpy())
    else:
        raise ValueError("Unsupported tensor dimension")
    df.to_csv(filename, index=False)

def load_tensor_from_csv(filename, shape):
    import pandas as pd
    df = pd.read_csv(filename)
    data = df.values
    tensor = torch.tensor(data)
    tensor = tensor.view(shape)
    return tensor
```
Unit test:
```
from loguru import logger
import random
import pytest
import torch
import ttnn
import traceback

from tests.ttnn.utils_for_testing import assert_with_pcc
aten = torch.ops.aten

    
def load_tensor_from_csv(filename, shape):
    import pandas as pd
    df = pd.read_csv(filename)
    data = df.values
    tensor = torch.tensor(data)
    tensor = tensor.view(shape)
    return tensor


def run_tanh_tests(
    file_path,
    input_shape, 
    dtype, 
    dlayout,
    device,
):
    x = load_tensor_from_csv(file_path[0], input_shape[0]).to(torch.bfloat16)

    try:
        # get ref result
        ref_value = aten.tanh.default(x, )
        
        tt_x = ttnn.from_torch(x, dtype=dtype[0], layout=dlayout[0], device=device)

        tt_result = ttnn.tanh(tt_x)
        tt_result = ttnn.to_torch(tt_result)
    except Exception as e:
        logger.warning(f"Test execution crashed: {e}")
        print(traceback.format_exc())
        raise e

    assert len(tt_result.shape) == len(ref_value.shape)
    assert tt_result.shape == ref_value.shape
    assert_with_pcc(ref_value, tt_result, 0.999)


test_sweep_args = [
    (
        ["path/to/tanh_input.csv"],
        [(1, 9, 3072)],
        [ttnn.bfloat16],
        [ttnn.TILE_LAYOUT],
    ),
]


@pytest.mark.parametrize(
    "file_path, input_shape, dtype, dlayout",
    (test_sweep_args),
)
def test_tanh(file_path, input_shape, dtype, dlayout, device):
    run_tanh_tests(file_path, input_shape, dtype, dlayout, device)

```

**Comment by ayerofieiev-tt** [comment]
@umadevimcw , can the team please take a look? Thank you!

**Comment by umadevimcw** [comment]
@ayerofieiev-tt Yeah will look at this
